src/main/java/io/appform/memq/HighLevelActor.java

```diff
@@ -0,0 +1,58 @@
+package io.appform.memq;
+
+
+import io.appform.memq.actor.Actor;
+import io.appform.memq.actor.ActorConfig;
+import io.appform.memq.actor.Message;
+import io.appform.memq.sideline.SidelineStore;
+import io.appform.memq.sideline.impl.NoOpSidelineStore;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Set;
+import java.util.function.ToIntFunction;
+
+@Slf4j
+public abstract class HighLevelActor<MessageType extends Enum<MessageType>, M extends Message> extends Actor<M> {
+
+    private final MessageType type;
+    private final SidelineStore<M> sidelineStore;
+
+    protected HighLevelActor(MessageType type,
+                             ActorConfig actorConfig,
+                             ActorSystem actorSystem,
+                             Set<Class<? extends Throwable>> ignoredErrors) {
+        this(type, actorConfig, actorSystem, ignoredErrors, null, null);
+    }
+
+    protected HighLevelActor(MessageType type,
+                             ActorConfig actorConfig,
+                             ActorSystem actorSystem,
+                             Set<Class<? extends Throwable>> ignoredErrors,
```

Comment:
Redundant .. this can be moved to a specialization of ExceptionHandler

Reply:
Done